src/DI/Definitions/FactoryDefinition.php

```diff
@@ -74,7 +74,7 @@ public function setResultDefinition(Definition $definition)
 
 
 	/**
-	 * @return ServiceDefinition
+	 * @return Definition|ServiceDefinition
```

Comment:
ServiceDefinition is Definition

Reply:
`@return ServiceDefinition` -> returns ServiceDefinition only

`@return Definition|ServiceDefinition` -> returns Definition or it's child ServiceDefinition. Definition only would make most sense here, but if we consider ServiceDefinition is almost everytime used then it's nice to satisfy IDE.

But I guess phpstan will start reporting invalid calls to methods which are on ServiceDefinition only. Imho better not to fix, just add it to ignore.

Reply:
What I mean is that it would be enough to write `Definition`.

And yes, it really exists only to satisfy IDE. But it is not correct, of course.

Reply:
Yes, I understand. I reverted old state.